src/reader.ts

```diff
@@ -0,0 +1,17 @@
+export interface Reader<T, U> {
+  map<K>(fn: (value: U) => K): Reader<T, K>,
```

Comment:
Please follow the style used in the other interfaces for object methods:
```typescript
interface Foo<U> {
  map: <K>(fn: (value: U) => K) => Foo<K>
}
```

commas are also not required in interfaces